src/ui/app/App.js

```diff
@@ -205,17 +206,10 @@ const RouteComponent = () => {
   const { COMPONENT_REPOSITORY_UI_ENABLED } = getRuntimeEnv();
   const contentMfe = useSelector(getMfeTargetContent);
 
-  const mfeRoutes = React.useMemo(() => contentMfe.reduce((acc, curr) => {
-    if (curr.paths) {
-      const routes = [];
-      curr.paths.forEach((route) => {
-        routes.push({ route, id: curr.id });
-      });
-      return [...acc, ...routes];
-    }
-    return acc;
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []), [JSON.stringify(contentMfe)]);
+  const mfeRoutes = React.useMemo(
+    () => generateMfeRoutes(contentMfe),
+    [contentMfe],
```

Comment:
this is an array of object, I think it should be stringified in order to correctly memoise it, what do u think @Dv92 ?